### PR TITLE
Ignore stderr when we call JavaFx provider first

### PR DESCRIPTION
**Pull request by yacaovsnc**
The stderr maybe benign, but in its presence our process fails.
We should first try parse the stdout and if we could parse out a
correct authorization code from the stdout, then treat the error
more seriously.  Otherwise even if there are stderr, but the code
is retrieved, we should simply ignore it.

### CODE REVIEW

**Comment by DavidStaheli** [comment]
Thanks, Yang.

File: oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java [hunk 1]
```diff
@@ -110,6 +108,18 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         catch (final InterruptedException e) {
             throw new AuthorizationException("interrupted_exception", e.getMessage(), null, e);
         }
+        catch (final AuthorizationException e) {
+            if ("parsing_error".equals(e.getCode()) && coordinator != null) {
```
Comment (olivierdagenais) [inline oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java]: It seems to me we could avoid filtering the `AuthorizationException` here by having provided the `errorContents` as a second parameter to `AuthorizationResponse#fromString` to initialize its `errorDescription` variable with.

File: oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java [hunk 2]
```diff
@@ -61,6 +61,10 @@ public static AuthorizationResponse fromString(final String s) throws Authorizat
             for (final String pair : pairs) {
                 final String[] nameAndValue = NAME_VALUE_SEPARATOR.split(pair, 2);
                 try {
+                    if (nameAndValue.length != 2) {
```
Comment (olivierdagenais) [inline oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java]: Do you have sample output you received from an OAuth 2.0 service that we could use to write a test with?

Comment (yacaovsnc) [inline oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java]: Nope, this fix isn't for what's returned from server.  This is actually to make the test work.  The unit test 'cut' returns something and when we try to parse it, we get an ArrayIndexOutOfBoundsException, this is to guard for that exception.  

I haven't seen a real life example that we would hit this yet.

File: oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java (hunk 1 above)
Comment (yacaovsnc) [inline oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java]: Can you please elaborate a little bit more on this?  

What I tried to do is there are two types of AuthorizationException that could throw from parsing the output now:
1. Parsing errors, somehow the stdout isn't really a name value pair
2. Legit server errors

So I don't want to hide the errorDescription if it's the second type.  Are you suggesting we could append to the errorContents to the errorDescription?

Comment (olivierdagenais) [inline oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java]: For the record, I opened pull request #15 to discuss and address this.

**Pull request merged**